app/code/Magento/InventoryConfiguration/Model/IsSourceItemManagementAllowedForProductType.php

```diff
@@ -33,6 +33,6 @@ public function __construct(StockConfigurationInterface $stockConfiguration)
      */
     public function execute(string $productType): bool
     {
-        return in_array($productType, array_keys(array_filter($this->stockConfiguration->getIsQtyTypeIds())), true);
+        return array_key_exists($productType, array_filter($this->stockConfiguration->getIsQtyTypeIds()));
```

Comment:
Previous implementation was type save. According to the documentation for `array_key_exists`, this function does not provide an type save option.
http://php.net/manual/de/function.array-key-exists.php

We need some unit test for this change in order to ensure, that type casting behaviour of `array_key_exists` does not run in any errors.

Reply:
I implemented a unit test and with that I tested an even more simple implementation (not yet pushed):
```
return $this->stockConfiguration->getIsQtyTypeIds()[$productType] ?? false;
```
What do you think?

Reply:
Hi @aleron75, I've added an comment to the test.

For this version of implementation `return $this->stockConfiguration->getIsQtyTypeIds()[$productType] ?? false;` we need to ensure, that the returned value is definitely a `boolean`, but this should work also :)

Reply:
Hi, @vadimjustus thank you for your reply.

Looking at the implementation it seems there is a bit of confusion.

Here is the method signature in `\Magento\CatalogInventory\Api\StockConfigurationInterface`
```
/**
 * @param int $filter
 * @return int[]
 */
public function getIsQtyTypeIds($filter = null);
```

From the above signature, I get that the returned array should contain `int` values.

But if I look at the concrete implementation in `\Magento\CatalogInventory\Model\Configuration`:
```
    /**
     * @param int|null $filter
     * @return array
     */
    public function getIsQtyTypeIds($filter = null)
    {
        if (null === $this->isQtyTypeIds) {
            $this->isQtyTypeIds = [];
            foreach ($this->config->getAll() as $typeId => $typeConfig) {
                $this->isQtyTypeIds[$typeId] = isset($typeConfig['is_qty']) ? $typeConfig['is_qty'] : false;
            }
        }
        $result = $this->isQtyTypeIds;
        if ($filter !== null) {
            foreach ($result as $key => $value) {
                if ($value !== $filter) {
                    unset($result[$key]);
                }
            }
        }
        return $result;
    }

```
So by looking at the concrete implementation, I can assume the returned array only contains boolean values but we have an inconsistency with docblock, in particular looking at:

```
$this->isQtyTypeIds[$typeId] = isset($typeConfig['is_qty']) ? $typeConfig['is_qty'] : false;
```

because by debugging I've checked that, if set, `$typeConfig['is_qty']` contains a boolean value.

If you agree I would update the docblock and commit the  changes proposed above.